src/calendar-heatmap.tsx

```diff
@@ -739,9 +738,10 @@ export class CalendarHeatmap extends Component<
       );
 
     // Add month data items to the overview
-    let item_block = this.items.selectAll('.item-block-month').data(month_data);
-    item_block.exit().remove();
-    item_block
+    this.items.selectAll('.item-block-month').exit().remove();
```

Comment:
With separate exit and remove it is causing the `selected elements` not to group. so `each` function is unable to run for the array.